### PR TITLE
Update UG

### PR DESCRIPTION
**Pull request by revdrag**
- UG Updates:
  - Add an `About` section
  - Separated `Features` into 3 subsections, namely `Class commands`, `Student commands` and `Miscellaneous commands`
  - Remove all coming soon features from `Table of Contents` and UG
  - Add a `Glossary` section
  - Update `FAQ` section
  - Update `Quick Start` section
  - Update add student and edit student features

- `EditStudentCommand` : Change the success message
  - Handle StudentNotFoundException by throwing an error message informing user to perform a `view` command first to retrieve the student list of the class
- `AddStudentCommand` : Change the success message

### CODE REVIEW

**Review by DonovanJJ — APPROVED** [review]
LGTM! Thanks for helping to sort out the formatting 👍

File: docs/UserGuide.md [hunk 1]
```diff
@@ -8,54 +8,126 @@ pageNav: 3
...
-    - [Adding a list of students : `coming in v1.3`](#adding-a-list-of-students--coming-in-v13)
-    - [Updating a student : `coming in v1.3`](#updating-a-student--coming-in-v13)
-    - [Adding a lesson : `coming in v1.3`](#adding-a-lesson--coming-in-v13)
-    - [Removing a lesson : `coming in v1.3`](#removing-a-lesson--coming-in-v13)
-    - [Finding a student : `coming in v1.3`](#finding-a-student--coming-in-v13)
+    - [Class Commands](#class-commands)
+      - [Adding a class : `add`](#adding-a-class-add)
+      - [Removing a class : `remove`](#removing-a-class-remove)
+      - [Viewing a class : `view`](#viewing-a-class-view)
+      - [Editing a class: `view`](#editing-a-class-edit)
```
Comment (songgthu) [inline docs/UserGuide.md]: I noticed a small typo here, it should be `edit` instead, can you help fix it too thanks!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -8,54 +8,126 @@ pageNav: 3
...
+      - [Starting a lesson : `startlesson`](#starting-a-lesson-startlesson)
+      - [Marking a student present : `mark`](#marking-your-students-attendance-as-present-mark)
+      - [Marking a student absent : `unmark`](#marking-all-students-attendance-in-your-class-as-present-markall)
+      - [Marking all students present : `markall`](#marking-your-students-attendance-as-absent-unmark)
+    - [Student commands](#student-commands)
+      - [Adding a student : `add /s`](#adding-a-student-add-s)
+      - [Removing a student : `remove`](#removing-a-student-remove)
+      - [Editing a student : `edit /s`](#editing-a-student-edit-s)
```
Comment (songgthu) [inline docs/UserGuide.md:40]: I think we should also standardize to include `/s` or not, for the class commands also

Comment (revdrag) [reply docs/UserGuide.md:40]: Okay!

**Review by seanpzk — APPROVED** [review]
LGTM!

**Pull request merged**